src/payment/strategies/cardinal/cardinal-client.ts

```diff
@@ -22,24 +22,33 @@ export interface CardinalOrderData {
 }
 
 export default class CardinalClient {
+    private _provider: string = '';
+    private _testMode: boolean = false;
     private _sdk?: Promise<CardinalSDK>;
-    private _isConfigured: boolean = false;
+    private _configurationToken: string = '';
 
     constructor(
         private _scriptLoader: CardinalScriptLoader
     ) {}
 
-    load(provider: string, testMode?: boolean): Promise<void> {
+    load(provider: string, testMode = false): Promise<void> {
+        this._provider = provider;
+        this._testMode = testMode;
+
         if (!this._sdk) {
             this._sdk = this._scriptLoader.load(provider, testMode);
         }
 
         return this._sdk.then(() => {});
     }
 
-    configure(clientToken: string): Promise<void> {
-        if (this._isConfigured) {
-            return Promise.resolve();
+    async configure(clientToken: string): Promise<void> {
```

Comment:
don't think you need `Promise` as return type if it's async

Reply:
Actually, it's not async. I was trying to do something and forgot to delete it. 🤦 Thank you 👍